### PR TITLE
DB-9686 Introduce new auto analyze mechanism

### PR DESCRIPTION
**Pull request by arnaud-splice**
## Short Description

Introduce a new auto analyze mechanism that kicks off analyze once a table has received a certain number of updates/deletes/inserts

## Long Description

This introduces a new column in SYS.SYSTABLES: `AUTO_ANALYZE`
This can be set per table or per schema via:
```
call syscs_util.set_auto_analyze('s', 't', 20)
```

Each region server maintains a count of modified rows. Once that count exceeds the auto_analyze value set in systables, an external connection is opened to trigger analyze on that table.
For now, this will only succeed if the user who ran the insertion has analyze privilege.

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2770/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2771/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2774/

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/DD_stats.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.splicemachine.db.catalog.UUID;
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.sql.dictionary.ConglomerateDescriptor;
+import com.splicemachine.db.iapi.sql.dictionary.DataDictionary;
+import com.splicemachine.db.iapi.sql.dictionary.TableDescriptor;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class DD_stats {
```
Comment (hatyo) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/DD_stats.java:26]: Name is not inline with our naming conventions

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/DMLWriteOperation.java [hunk 2]
```diff
@@ -500,15 +519,55 @@ public void openCore() throws StandardException {
...
+                RemoteUser.setRemoteUser(user);
+                try (Connection connection = RemoteUser.getConnection("Arnauds-MacBook-Pro.local:1528");
+                     Statement stmt = connection.createStatement()) {
+                    stmt.execute(String.format("analyze table %s", tableQualifiedName));
+                } catch (SQLException e) {
+                    SpliceLogUtils.error(LOG, "Automatic ANALYZE of %s failed: %s", tableQualifiedName, e);
+                }
+            }
         }
+        Thread thread = new Thread(new AnalyzeRunnable(RemoteUser.user.get()));
```
Comment (hatyo) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/DMLWriteOperation.java:569]: Please name the thread, it makes easier to debug

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/InsertOperation.java [hunk 3]
```diff
@@ -372,7 +372,9 @@ else if (storedAs.toLowerCase().equals("t"))
                     .execRowDefinition(getExecRowDefinition())
                     .loadReplaceMode(loadReplaceMode)
                     .build();
-            return writer.write();
+
+            DataSet<ExecRow> written = writer.write();
```
Comment (hatyo) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/InsertOperation.java:376]: please revert

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/DMLWriteOperation.java [hunk 4]
```diff
@@ -500,15 +519,55 @@ public void openCore() throws StandardException {
...
+    private void runAnalyzeInOwnThread(String tableQualifiedName) throws SQLException {
+        class AnalyzeRunnable implements Runnable {
+            private RemoteUser user;
+            private AnalyzeRunnable(RemoteUser user) {
+                this.user = user;
+            }
+            public void run() {
+                SpliceLogUtils.info(LOG, "Automatically running ANALYZE for %s", tableQualifiedName);
+                RemoteUser.setRemoteUser(user);
+                try (Connection connection = RemoteUser.getConnection("Arnauds-MacBook-Pro.local:1528");
```
Comment (hatyo) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/DMLWriteOperation.java:561]: I think this is incorrect, please remove/refactor.

**Review by hatyo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**